### ISSUE TITLE
[Bug]: `plt.imsave` return all white image,but `plt.imshow` is correct 

### ISSUE THREAD

**Issue by danyow-cheung** [matplotlib/matplotlib#25653]
### Bug summary

I am doing a MelSpectrogram and I want to save the images to my disk ,but the `plt.imsave` can't not save the correct image. In return, i got all white image.

my plt version : 3.7.1 

### Code for reproduction

```python
from torch.utils.data import Dataset,dataloader
from pathlib import  Path
import os 

def load_audio_files(path):
    walker = sorted(str(p) for p in Path(path).glob(f'*.wav'))
    for i,file_path in enumerate(walker):
        
        waveform,sample_rate = torchaudio.load(file_path)

        mel_spectrogram = torchaudio.transforms.MelSpectrogram(sample_rate)(waveform)
        plt.imshow(mel_spectrogram.log2()[0,:,:].numpy())
        plt.imsave(f'data/car/spectrograms/{i}-s.jpg',mel_spectrogram.log2()[0,:,:].numpy())
        
        if i==5:
            break

load_audio_files(os.path.join(cur_path,'data/car/audio/'))
```

### Actual outcome

![捕获](https://user-images.githubusercontent.com/76671016/230854703-4f739bf4-20a3-444c-9a1a-3516d2c8d681.PNG)

### Expected outcome

![捕获1](https://user-images.githubusercontent.com/76671016/230854802-700200a1-9342-4643-8043-614e7fbd43ae.PNG)

### Operating system

window10

### Matplotlib Version

3.7.1 

### Python version

3.9.10

### Installation

pip

**Comment by jklymak** [comment]
Can you reproduce without torch and external data?  Otherwise we cannot reproduce.  Thanks

**Comment by danyow-cheung** [comment]
> Can you reproduce without torch and external data? Otherwise we cannot reproduce. Thanks

yes, i load image file through `Image.open` and `plt.imsave`,here is the result 
![捕获](https://user-images.githubusercontent.com/76671016/231036347-abf00c56-49ab-4974-96cd-455388fa2f01.PNG)

**Comment by jklymak** [comment]
Still cannot reproduce w/o `spec_img0.png` - most importantly what its format is (RGB, RGBA, or greyscale).

**Comment by rcomer** [comment]
Is this the same as #24610?
